manifests/plugin/container_gateway.pp

```diff
@@ -0,0 +1,39 @@
+# = Foreman Proxy Container Gateway plugin
+#
+# This class installs the Container Gateway plugin
+#
+# $pulp_endpoint::            Pulp 3 server endpoint
+#
+# $pulp_client_ssl_cert::     X509 certificate for authenticating with Pulp
+#
+# $pulp_client_ssl_key::      RSA private key for the Pulp certificate
+#
+# $sqlite_db_path::           Absolute path for the SQLite DB file to exist at
+#
+# === Advanced parameters:
+#
+# $enabled::                  enables/disables the pulp plugin
+#
+# $listen_on::                proxy feature listens on http, https, or both
+#
+# $version::                  plugin package version, it's passed to ensure parameter of package resource
+#                             can be set to specific version number, 'latest', 'present' etc.
+#
+class foreman_proxy::plugin::container_gateway (
+  Optional[String] $version = $foreman_proxy::plugin::container_gateway::params::version,
+  Boolean $enabled = $foreman_proxy::plugin::container_gateway::params::enabled,
+  Foreman_proxy::ListenOn $listen_on = $foreman_proxy::plugin::container_gateway::params::listen_on,
+  Stdlib::HTTPUrl $pulp_endpoint = $foreman_proxy::plugin::container_gateway::params::pulp_endpoint,
+  Stdlib::Absolutepath $pulp_client_ssl_cert = $foreman_proxy::plugin::container_gateway::params::pulp_ssl_client_cert,
+  Stdlib::Absolutepath $pulp_client_ssl_key = $foreman_proxy::plugin::container_gateway::params::pulp_ssl_client_key,
+  Stdlib::Absolutepath $sqlite_db_path = $foreman_proxy::plugin::container_gateway::params::sqlite_db_path,
```

Comment:
I'm a bit confused, so if `$pulp_endpoint` isn't declared here, where would it go? It looks like I'm doing pretty much the same thing as`$pulp_url` in the Pulp plugin for example.

Reply:
So a bit more insight from my point. Now that I've seen the various PRs I changed my opinion a bit. If we're not going to expose it via Kafo, there really isn't any point in keeping `params.pp`.

There are 2 reasons to have `params.pp`:
* You need logic (if/else/...) to determine the value, like OS-dependent defaults
* You want to expose the class via Kafo and need something Kafo can't parse

Let's start with the last part. We expose the defaults via [puppet-strings](https://github.com/puppetlabs/puppet-strings). This gives the raw Puppet code as a default. Then [kafo_parsers](https://github.com/theforeman/kafo_parsers) has some [very naive "parsing"](https://github.com/theforeman/kafo_parsers/blob/b9a0432e140464d62c440fec94708d6a6db8fb53/lib/kafo_parsers/puppet_strings_module_parser.rb#L152-L164). This naive parsing can't deal with a lot of things, like `"https://${facts['networking']['fqdn']}"` but it can deal with `$variable`. Putting the default value in a `params.pp` is thus a workaround, but it works.

For the second part, you can also use data in modules which utilizes hiera, but we haven't really used that and Kafo doesn't understand it either so let's leave that out of scope now.

Reply:
Alright, thanks for the explanation. I assumed at firs that Kafo was something that was being used in the background rather than something I to purposefully use.  I'll remove `params.pp` and move the defaults into `container_gateway.pp`.